### PR TITLE
move some common code to echidna-ruby

### PR DESCRIPTION
**Pull request by flyerhzm**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -11,6 +8,10 @@ gem 'activesupport', require: 'active_support/all'
 gem 'oauth2', git: 'git://github.com/rainux/oauth2'
 gem 'tencent-weibo', git: 'git://github.com/rainux/tencent-weibo'
 
+gem 'echidna-ruby', git: 'git@github.com:transist/echidna-ruby.git',
+                    require: ['echidna/path', 'echidna/logger', 'echidna/redis', 'echidna/dependency']
```
Comment (0x1b2c) [inline Gemfile]: For easier usage, I'd like to suggest create a `lib/echidna-ruby.rb` (same name with the gem) file to require all other files, so we can remove this long require option for all apps.

**Comment by 0x1b2c** [comment]
Also we may move `lib/redis/*.rb` to `echidna-ruby`, how do you think?

File: Gemfile (hunk 1 above)
Comment (flyerhzm) [inline Gemfile]: yes, I'm considering adding it.

On Sat, Mar 2, 2013 at 7:19 PM, Rainux Luo notifications@github.com wrote:

> In Gemfile:
> 
> > @@ -11,6 +8,10 @@ gem 'activesupport', require: 'active_support/all'
> >  gem 'oauth2', git: 'git://github.com/rainux/oauth2'
> >  gem 'tencent-weibo', git: 'git://github.com/rainux/tencent-weibo'
> > 
> > +gem 'echidna-ruby', git: 'git@github.com:transist/echidna-ruby.git',
> > -                    require: ['echidna/path', 'echidna/logger', 'echidna/redis', 'echidna/dependency']
> 
> For easier usage, I'd like to suggest create a lib/echidna-ruby.rb (same
> name with the gem) file to require all other files, so we can remove this
> long require option for all apps.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/transist/echidna-spider/pull/11/files#r3214809
> .

## 

Best regards,
Richard Huang

http://huangzhimin.com
http://www.huangzhimin.comhttp://rails-bestpractices.com,
http://railsbp.com
https://github.com/flyerhzm http://github.com/flyerhzm,
https://github.com/railsbp
@flyerhzm, @railsbp

**Comment by flyerhzm** [comment]
sure, feel free to move it.

On Sat, Mar 2, 2013 at 7:21 PM, Rainux Luo notifications@github.com wrote:

> Also we may move lib/redis/redis_model.rb to echidna-ruby, how do think?
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/transist/echidna-spider/pull/11#issuecomment-14326744
> .

## 

Best regards,
Richard Huang

http://huangzhimin.com
http://www.huangzhimin.comhttp://rails-bestpractices.com,
http://railsbp.com
https://github.com/flyerhzm http://github.com/flyerhzm,
https://github.com/railsbp
@flyerhzm, @railsbp

**Comment by flyerhzm** [comment]
<http://www.skillpages.com/i/ftuf/emailinvitelanding?user=bb17c8cc-be86-e211-bb20
-b4fd0eafa145&signedParameters=mu_28T5J6_kDa0ixdNoYs9l-BLSTK6j-Q14usN0XC_E.eyJlbW
FpbCI6InJlcGx5K2ktMTE1NzQ4NDgtNTI1MmUwNjhjYjUzNWM1NjdhMTI5NDc0ZTViMjYyN2UwODUzZjB
hMy02NjgzNkByZXBseS5naXRodWIuY29tIiwiZnJvbSI6ImJiMTdjOGNjLWJlODYtZTIxMS1iYjIwLWI0
ZmQwZWFmYTE0NSIsInRvIjoiYzQxZjVmOGMtYjE5My01M2NmLWFmNzctMzI4ODA3ZTQ5MzBmIiwiaXNzd
WVkX2F0IjoiMTM2MzgyNDQxMCIsInZlciI6IjEiLCJwIjoiMyJ9&emailMessageId=14e119d5-ba91-
e211-bbf5-8d1ec1186671&emailType=InviteSecondReminder&sdate=201303210006&utm_sour
ce=EmailInviteSecondReminder&utm_medium=Email&emailLinkType=Logo>

Hi,

Richard Huang is still waiting to connect with you.

<http://www.skillpages.com/i/ftuf/emailinvitelanding?user=bb17c8cc-be86-e211-bb20
-b4fd0eafa145&signedParameters=mu_28T5J6_kDa0ixdNoYs9l-BLSTK6j-Q14usN0XC_E.eyJlbW
FpbCI6InJlcGx5K2ktMTE1NzQ4NDgtNTI1MmUwNjhjYjUzNWM1NjdhMTI5NDc0ZTViMjYyN2UwODUzZjB
hMy02NjgzNkByZXBseS5naXRodWIuY29tIiwiZnJvbSI6ImJiMTdjOGNjLWJlODYtZTIxMS1iYjIwLWI0
ZmQwZWFmYTE0NSIsInRvIjoiYzQxZjVmOGMtYjE5My01M2NmLWFmNzctMzI4ODA3ZTQ5MzBmIiwiaXNzd
WVkX2F0IjoiMTM2MzgyNDQxMCIsInZlciI6IjEiLCJwIjoiMyJ9&emailMessageId=14e119d5-ba91-
e211-bbf5-8d1ec1186671&emailType=InviteSecondReminder&sdate=201303210006&utm_sour
ce=EmailInviteSecondReminder&utm_medium=Email&emailLinkType=ProfileThumb>

Richard Huang
<http://www.skillpages.com/i/ftuf/emailinvitelanding?user=bb17c8cc-be86-e211-bb20
-b4fd0eafa145&signedParameters=mu_28T5J6_kDa0ixdNoYs9l-BLSTK6j-Q14usN0XC_E.eyJlbW
FpbCI6InJlcGx5K2ktMTE1NzQ4NDgtNTI1MmUwNjhjYjUzNWM1NjdhMTI5NDc0ZTViMjYyN2UwODUzZjB
hMy02NjgzNkByZXBseS5naXRodWIuY29tIiwiZnJvbSI6ImJiMTdjOGNjLWJlODYtZTIxMS1iYjIwLWI0
ZmQwZWFmYTE0NSIsInRvIjoiYzQxZjVmOGMtYjE5My01M2NmLWFmNzctMzI4ODA3ZTQ5MzBmIiwiaXNzd
WVkX2F0IjoiMTM2MzgyNDQxMCIsInZlciI6IjEiLCJwIjoiMyJ9&emailMessageId=14e119d5-ba91-
e211-bbf5-8d1ec1186671&emailType=InviteSecondReminder&sdate=201303210006&utm_sour
ce=EmailInviteSecondReminder&utm_medium=Email&emailLinkType=ProfileName>

Invite sent: 7 March, 2013

Continue
<http://www.skillpages.com/i/ftuf/emailinvitelanding?user=bb17c8cc-be86-e211-bb20
-b4fd0eafa145&signedParameters=mu_28T5J6_kDa0ixdNoYs9l-BLSTK6j-Q14usN0XC_E.eyJlbW
FpbCI6InJlcGx5K2ktMTE1NzQ4NDgtNTI1MmUwNjhjYjUzNWM1NjdhMTI5NDc0ZTViMjYyN2UwODUzZjB
hMy02NjgzNkByZXBseS5naXRodWIuY29tIiwiZnJvbSI6ImJiMTdjOGNjLWJlODYtZTIxMS1iYjIwLWI0
ZmQwZWFmYTE0NSIsInRvIjoiYzQxZjVmOGMtYjE5My01M2NmLWFmNzctMzI4ODA3ZTQ5MzBmIiwiaXNzd
WVkX2F0IjoiMTM2MzgyNDQxMCIsInZlciI6IjEiLCJwIjoiMyJ9&cnId=bb17c8cc-be86-e211-bb20-
b4fd0eafa145&emailMessageId=14e119d5-ba91-e211-bbf5-8d1ec1186671&emailType=Invite
SecondReminder&sdate=201303210006&utm_source=EmailInviteSecondReminder&utm_medium
=Email&emailLinkType=MainCallToAction>

reply@reply.github.com
was invited to join SkillPages by Richard Huang. To stop receiving emails from
SkillPages click here
<http://www.skillpages.com/i/account/unsubscribe?signedParameters=xr8LLZWWZtLfyTv
V7ii0TCsTtcM77xNAb6Iz3KFtyuI.eyJlbWFpbCI6InJlcGx5K2ktMTE1NzQ4NDgtNTI1MmUwNjhjYjUz
NWM1NjdhMTI5NDc0ZTViMjYyN2UwODUzZjBhMy02NjgzNkByZXBseS5naXRodWIuY29tIiwiaXNzdWVkX
2F0IjoiMTM2MzgyNDQxMCIsInZlciI6IjEiLCJwIjoiMCJ9&emailMessageId=14e119d5-ba91-e211
-bbf5-8d1ec1186671&emailType=InviteSecondReminder&sdate=201303210006&utm_source=E
mailInviteSecondReminder&utm_medium=Email&emailLinkType=Unsubscribe> .

© 2013 SkillPages, Blackrock Business Park, Dublin, Ireland and 228 Hamilton
Avenue, 3rd Floor, Palo Alto, CA 94301.

File: Gemfile (hunk 1 above)
Comment (flyerhzm) [inline Gemfile]: <http://www.skillpages.com/i/ftuf/emailinvitelanding?user=bb17c8cc-be86-e211-bb20
-b4fd0eafa145&signedParameters=o1g4IhznBnVvv-dUdQlvnE67yHgyiqxsZtsxp16RQIU.eyJlbW
FpbCI6InJlcGx5K3AtMzIxNDgwOS0zYmY0YjUyOTZhYzVkOTdiYWMyMGNmMTcwNDBiNWFjMjIyZDMzZmZ
kLTY2ODM2QHJlcGx5LmdpdGh1Yi5jb20iLCJmcm9tIjoiYmIxN2M4Y2MtYmU4Ni1lMjExLWJiMjAtYjRm
ZDBlYWZhMTQ1IiwidG8iOiI0MjJkYWQ2ZC02NTFkLTVlZjktODdmNy00MWYxNTUwYjhjMjUiLCJpc3N1Z
WRfYXQiOiIxMzYzODI0NDE1IiwidmVyIjoiMSIsInAiOiIzIn0&emailMessageId=99e019d5-ba91-e
211-bbf5-8d1ec1186671&emailType=InviteSecondReminder&sdate=201303210006&utm_sourc
e=EmailInviteSecondReminder&utm_medium=Email&emailLinkType=Logo>

Hi,

Richard Huang is still waiting to connect with you.

<http://www.skillpages.com/i/ftuf/emailinvitelanding?user=bb17c8cc-be86-e211-bb20
-b4fd0eafa145&signedParameters=o1g4IhznBnVvv-dUdQlvnE67yHgyiqxsZtsxp16RQIU.eyJlbW
FpbCI6InJlcGx5K3AtMzIxNDgwOS0zYmY0YjUyOTZhYzVkOTdiYWMyMGNmMTcwNDBiNWFjMjIyZDMzZmZ
kLTY2ODM2QHJlcGx5LmdpdGh1Yi5jb20iLCJmcm9tIjoiYmIxN2M4Y2MtYmU4Ni1lMjExLWJiMjAtYjRm
ZDBlYWZhMTQ1IiwidG8iOiI0MjJkYWQ2ZC02NTFkLTVlZjktODdmNy00MWYxNTUwYjhjMjUiLCJpc3N1Z
WRfYXQiOiIxMzYzODI0NDE1IiwidmVyIjoiMSIsInAiOiIzIn0&emailMessageId=99e019d5-ba91-e
211-bbf5-8d1ec1186671&emailType=InviteSecondReminder&sdate=201303210006&utm_sourc
e=EmailInviteSecondReminder&utm_medium=Email&emailLinkType=ProfileThumb>

Richard Huang
<http://www.skillpages.com/i/ftuf/emailinvitelanding?user=bb17c8cc-be86-e211-bb20
-b4fd0eafa145&signedParameters=o1g4IhznBnVvv-dUdQlvnE67yHgyiqxsZtsxp16RQIU.eyJlbW
FpbCI6InJlcGx5K3AtMzIxNDgwOS0zYmY0YjUyOTZhYzVkOTdiYWMyMGNmMTcwNDBiNWFjMjIyZDMzZmZ
kLTY2ODM2QHJlcGx5LmdpdGh1Yi5jb20iLCJmcm9tIjoiYmIxN2M4Y2MtYmU4Ni1lMjExLWJiMjAtYjRm
ZDBlYWZhMTQ1IiwidG8iOiI0MjJkYWQ2ZC02NTFkLTVlZjktODdmNy00MWYxNTUwYjhjMjUiLCJpc3N1Z
WRfYXQiOiIxMzYzODI0NDE1IiwidmVyIjoiMSIsInAiOiIzIn0&emailMessageId=99e019d5-ba91-e
211-bbf5-8d1ec1186671&emailType=InviteSecondReminder&sdate=201303210006&utm_sourc
e=EmailInviteSecondReminder&utm_medium=Email&emailLinkType=ProfileName>

Invite sent: 7 March, 2013

Continue
<http://www.skillpages.com/i/ftuf/emailinvitelanding?user=bb17c8cc-be86-e211-bb20
-b4fd0eafa145&signedParameters=o1g4IhznBnVvv-dUdQlvnE67yHgyiqxsZtsxp16RQIU.eyJlbW
FpbCI6InJlcGx5K3AtMzIxNDgwOS0zYmY0YjUyOTZhYzVkOTdiYWMyMGNmMTcwNDBiNWFjMjIyZDMzZmZ
kLTY2ODM2QHJlcGx5LmdpdGh1Yi5jb20iLCJmcm9tIjoiYmIxN2M4Y2MtYmU4Ni1lMjExLWJiMjAtYjRm
ZDBlYWZhMTQ1IiwidG8iOiI0MjJkYWQ2ZC02NTFkLTVlZjktODdmNy00MWYxNTUwYjhjMjUiLCJpc3N1Z
WRfYXQiOiIxMzYzODI0NDE1IiwidmVyIjoiMSIsInAiOiIzIn0&cnId=bb17c8cc-be86-e211-bb20-b
4fd0eafa145&emailMessageId=99e019d5-ba91-e211-bbf5-8d1ec1186671&emailType=InviteS
econdReminder&sdate=201303210006&utm_source=EmailInviteSecondReminder&utm_medium=
Email&emailLinkType=MainCallToAction>

reply@reply.github.com
was invited to join SkillPages by Richard Huang. To stop receiving emails from
SkillPages click here
<http://www.skillpages.com/i/account/unsubscribe?signedParameters=aHGusP_WLZBLbdW
ukiZkHIg6wDHC4hTwGN5pO3nO304.eyJlbWFpbCI6InJlcGx5K3AtMzIxNDgwOS0zYmY0YjUyOTZhYzVk
OTdiYWMyMGNmMTcwNDBiNWFjMjIyZDMzZmZkLTY2ODM2QHJlcGx5LmdpdGh1Yi5jb20iLCJpc3N1ZWRfY
XQiOiIxMzYzODI0NDE1IiwidmVyIjoiMSIsInAiOiIwIn0&emailMessageId=99e019d5-ba91-e211-
bbf5-8d1ec1186671&emailType=InviteSecondReminder&sdate=201303210006&utm_source=Em
ailInviteSecondReminder&utm_medium=Email&emailLinkType=Unsubscribe> .

© 2013 SkillPages, Blackrock Business Park, Dublin, Ireland and 228 Hamilton
Avenue, 3rd Floor, Palo Alto, CA 94301.

**Pull request merged**